### ISSUE TITLE
SSR PoweredBy

### ISSUE THREAD

**Issue by davidfurlong** [algolia/react-instantsearch#178]
**Do you want to request a *feature* or report a *bug*?**

bug

**Bug: What is the current behavior?**

Trying to SSR `findResultsState` with included PoweredBy Component => `ReferenceError: location is not defined`

https://github.com/algolia/react-instantsearch/blob/master/packages/react-instantsearch/src/connectors/connectPoweredBy.js#L20

**Bug: What is the expected behavior?**

 Check if on the server so you dont try to use global location 👍 

**What is the version you are using? Always use the latest one before opening a bug issue.**

 "react-instantsearch": "^4.1.0-beta.1",

Additionally the SSR guide seems overly simplistic. 
Doing: `  const resultsState = await findResultsState(App);` which does a full render of App on the server will not work well if App is not your root component. However using your root component will provide the resultsState to props on the wrong component. I got it working by writing my own InstantSearchProvider, withResults HOCs and mocking my other contexts and props for the component invoked in findResultsState. However I did get it working and thats 💯  because SSR is super important for us

Also I hope you don't mind I temporarily remove this component :)

**Comment by mthuret** [comment]
Thanks for the feedback @davidfurlong!

We'll fix the poweredBy. 

About the guide, I'll try because I thought it was possible to have it not as the root component but by forwarding correctly the props. I'll try again and let you know. And if you want to share some code, feel free it's awesome if you make it works for your use case :)

**Comment by davidfurlong** [comment]
Relevant additional changes:

```js
const AlgoliaRenderSearchViewLite = withContext(
  {
    debug: PropTypes.bool.isRequired,
    g: PropTypes.func.isRequired,
    locale: PropTypes.func.isRequired,
    subscriptions: PropTypes.object.isRequired,
    setLocale: PropTypes.func.isRequired,
    store: PropTypes.object.isRequired,
  },
  props => ({
    store: { getState: () => {} },
    locale: () => 'en',
    g: () => '',
    debug: false,
    setLocale: () => null,
    subscriptions: {
      subscribe: () => null,
    },
  })
)(SearchViewLite);
```

```js
import React from 'react';
import PropTypes from 'prop-types';
 
const ConnectInstantSearchResultsState = WrappedComponent => {
  class InstantSearchResultsState extends React.Component {
    render(){
      return <WrappedComponent {...this.props} resultsState={this.context.resultsState} />
    }
  }
  InstantSearchResultsState.contextTypes = {
    resultsState: PropTypes.any
  }
  return InstantSearchResultsState;
}
 
export default ConnectInstantSearchResultsState;
```

```js
import { Component, Children } from 'react';
import PropTypes from 'prop-types';
 
export function createProvider(storeKey = 'algolia-ssr', subKey) {
  class Provider extends Component {
    getChildContext() {
      return { [storeKey]: this[storeKey] };
    }
 
    constructor(props, context) {
      super(props, context);
      this[storeKey] = props.resultsState;
    }
 
    render() {
      return Children.only(this.props.children);
    }
  }
 
  Provider.propTypes = {
    resultsState: PropTypes.any.isRequired,
    children: PropTypes.element.isRequired,
  };
  Provider.childContextTypes = {
    [storeKey]: PropTypes.any.isRequired,
  };
 
  return Provider;
}
 
export default createProvider();
```

**Comment by davidfurlong** [comment]
Additionally it doesn't play nice with react-router so you need to manually only do this duplicate deep render on the routes you need it on

**Comment by mthuret** [comment]
Fix with 4.1.0.beta3
